src/app/modules/form/public-form/public-form.controller.ts

```diff
@@ -485,33 +486,45 @@ export const handleFormAuthRedirect = [
 
 /**
  * NOTE: This is exported only for testing
- * Logs user out of SP / CP By deleting cookie
+ * Logs user out of SP / CP / MyInfo / SGID by deleting cookie
  * @param authType type of authentication
  *
  * @returns 200 with success message when user logs out successfully
  * @returns 400 if authType is invalid
  */
 export const _handleSpcpLogout: ControllerHandler<
-  { authType: AuthType.SP | AuthType.CP | AuthType.SGID },
+  { authType: AuthType.SP | AuthType.CP | AuthType.MyInfo | AuthType.SGID },
   PublicFormAuthLogoutDto
 > = (req, res) => {
   const { authType } = req.params
 
-  return res
-    .clearCookie(JwtName[authType])
-    .status(200)
-    .json({ message: 'Successfully logged out.' })
+  if (authType === AuthType.MyInfo) {
+    return res
+      .clearCookie(MYINFO_COOKIE_NAME)
+      .status(200)
+      .json({ message: 'Successfully logged out.' })
+  } else if (authType === AuthType.SGID) {
+    return res
+      .clearCookie(SGID_COOKIE_NAME)
+      .status(200)
+      .json({ message: 'Successfully logged out.' })
+  } else {
+    return res
```

Comment:
if you're really not keen on a switch statement, can we at least take this out of the `else` block so that there is a return guaranteed? I still a switch is cleaner since you're repeating the return statements anyway:
```
switch (authType) {
  case AuthType.MyInfo:
    return res
      .clearCookie(MYINFO_COOKIE_NAME)
      .status(200)
      .json({ message: 'Successfully logged out.' })
  case AuthType.SGID:
    return res
      .clearCookie(SGID_COOKIE_NAME)
      .status(200)
      .json({ message: 'Successfully logged out.' })
  default:
    return res
      .clearCookie(JwtName[authType])
      .status(200)
      .json({ message: 'Successfully logged out.' })
}
```

Reply:
........................ok la but this doesn't guarantee a return statement also LOL